src/System Application/App/Document Processor/src/PDFDocumentProcessorImpl.Codeunit.al

```diff
@@ -0,0 +1,68 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.IO;
+
+using System.Utilities;
+using System;
+
+/// <summary>
+/// Codeunit that provides helper functions for PDF processing.
+/// </summary>
+codeunit 3109 "PDF Document Processor Impl."
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    var
+        UnsupportedImageFormatErr: Label 'Unsupported image format: %1', Comment = '%1 is the image format that is not supported.';
+
+    procedure ConvertToImage(DocumentStream: InStream; var ImageStream: InStream; ImageFormat: Enum "Image Format"; PageNumber: Integer): Boolean
+    var
+        PdfConverterInstance: DotNet PdfConverter;
+        PdfTargetDevice: DotNet PdfTargetDevice;
+        MemoryStream: DotNet MemoryStream;
+        ImageMemoryStream: DotNet MemoryStream;
+        SharedDocumentStream: InStream;
+    begin
+        // Empty stream, no actions possible on the stream so return immediately
+        if DocumentStream.Length() < 1 then
```

Comment:
Related to previous comment, we have no such checks in the product as far as I can see. This is the first time we check if it is less than 1. I'm not sure if it is correct. It can be -1 if length cannot be determined, I'm not sure when it is raised.

Reply:
Im all ears. How else would you determine it? There sis no IsEmpty, there is no blob. This is the only method that says something about the content size.